src/data/move.ts

```diff
@@ -2875,6 +2875,95 @@ export class StatStageChangeAttr extends MoveEffectAttr {
   }
 }
 
+export class SecretPowerAttr extends MoveEffectAttr {
+  constructor() {
+    super(false);
+  }
+
+  apply(user: Pokemon, target: Pokemon, move: Move, args?: any[] | undefined): boolean | Promise<boolean> {
+    let secondaryEffect: MoveAttr;
```

Comment:
```suggestion
  apply(user: Pokemon, target: Pokemon, move: Move, args?: any[]): boolean | Promise<boolean> {
    let secondaryEffect: MoveEffectAttr;
```
Not sure it matters too much, but all the invoked attrs extend `MoveEffectAttr`.